lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services
+  attr_accessor :registered_vehicles,
+                :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(facility_details)
+    @name                 = facility_details[:name]
+    @address              = facility_details[:address]
+    @phone                = facility_details[:phone]
+    @services             = []
+    @registered_vehicles  = []
+    @collected_fees       = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+      vehicle.registration_date = Date.today
+      if vehicle.antique?
+        vehicle.plate_type = :antique
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle?
+        vehicle.plate_type = :ev
+        @collected_fees += 200
+      else 
+        vehicle.plate_type = :regular
+        @collected_fees += 100
+      end
+      @registered_vehicles << vehicle
+    else
+      nil
+    end
+  end
```

Comment:
`register_vehicle` is doing a lot. I might try to pull out the different functions happening here to their own methods. Something like:
```ruby
  def register_vehicle(vehicle)
    if @services.include?("Vehicle Registration")
      collect_fees(vehicle)
      set_plate_type(vehicle)
      set_registration_date(vehicle)
      @registered_vehicles << vehicle
    end
  end

  def collect_fees(vehicle)
    if vehicle.antique?
      @collected_fees += 25
    elsif vehicle.electric_vehicle?
      @collected_fees += 200
    else
      @collected_fees += 100
    end
  end

  def set_registration_date(vehicle)
    vehicle.registration_date = Date.today
  end

  def set_plate_type(vehicle)
    if vehicle.antique?
      vehicle.plate_type = :antique
    elsif vehicle.electric_vehicle?
      vehicle.plate_type = :ev
    else
      vehicle.plate_type = :regular
    end
  end
```